src/java.base/share/classes/sun/security/ssl/X509TrustManagerImpl.java

```diff
@@ -428,8 +428,17 @@ static void checkIdentity(SSLSession session,
         }
 
         if (!identifiable) {
-            checkIdentity(peerHost,
-                    trustedChain[0], algorithm, chainsToPublicCA);
+            try {
```

Comment:
Minor formatting nits/suggestions:
```
if(
->
if (
```

Lines <= 80 chars, please.

```
Endpoint Identification algorithm
->
Endpoint Identification Algorithm
```

Reply:
Done, attached is the test result 
[https_endpt.zip](https://github.com/user-attachments/files/16139516/https_endpt.zip)